### ISSUE TITLE
Page.getPlainText broken - PlainTextConverter struggles to discriminate candidate methods and ends in 'VisitorException'

### ISSUE THREAD

**Issue by mawiesne** [dkpro/dkpro-jwpl#160]
With the introduction of Swebble 3.1.7 to the JWPL 1.2.0-SNAPSHOT line, I can no longer fetch plain text data from Wikipedia backends via `Page.getPlainText`. The stacktrace is documented here:

```
ERROR - de.fau.cs.osr.utils.visitor.VisitorException: vClass: de.tudarmstadt.ukp.wikipedia.api.sweble.PlainTextConverter
nClass: org.sweble.wikitext.parser.nodes.WtText
Candidate 1: visit(org.sweble.wikitext.parser.nodes.WtNode)
Candidate 2: visit(de.fau.cs.osr.ptk.common.ast.AstText)

de.fau.cs.osr.utils.visitor.VisitingException: de.fau.cs.osr.utils.visitor.VisitorException: vClass: de.tudarmstadt.ukp.wikipedia.api.sweble.PlainTextConverter
nClass: org.sweble.wikitext.parser.nodes.WtText
Candidate 1: visit(org.sweble.wikitext.parser.nodes.WtNode)
Candidate 2: visit(de.fau.cs.osr.ptk.common.ast.AstText)

	at de.fau.cs.osr.utils.visitor.VisitorBase.handleVisitingException(VisitorBase.java:92)
	at de.fau.cs.osr.utils.visitor.VisitorLogic.resolveAndVisit(VisitorLogic.java:118)
	at de.fau.cs.osr.utils.visitor.VisitorLogic.resolveAndVisit(VisitorLogic.java:90)
	at de.fau.cs.osr.utils.visitor.VisitorBase.resolveAndVisit(VisitorBase.java:119)
	at de.fau.cs.osr.ptk.common.AstVisitor.dispatch(AstVisitor.java:56)
	at de.fau.cs.osr.ptk.common.AstVisitor.iterate(AstVisitor.java:66)
	at de.tudarmstadt.ukp.wikipedia.api.sweble.PlainTextConverter.visit(PlainTextConverter.java:189)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at de.fau.cs.osr.utils.visitor.VisitorLogic$Target.invoke(VisitorLogic.java:361)
	at de.fau.cs.osr.utils.visitor.VisitorLogic.resolveAndVisit(VisitorLogic.java:110)
	at de.fau.cs.osr.utils.visitor.VisitorLogic.resolveAndVisit(VisitorLogic.java:90)
	at de.fau.cs.osr.utils.visitor.VisitorBase.resolveAndVisit(VisitorBase.java:119)
	at de.fau.cs.osr.ptk.common.AstVisitor.dispatch(AstVisitor.java:56)
	at de.fau.cs.osr.ptk.common.AstVisitor.dispatch(AstVisitor.java:28)
	at de.fau.cs.osr.utils.visitor.VisitorBase.go(VisitorBase.java:111)
	at de.tudarmstadt.ukp.wikipedia.api.Page.parsePage(Page.java:610)
	at de.tudarmstadt.ukp.wikipedia.api.Page.getPlainText(Page.java:591)
	at ...
	at java.lang.Thread.run(Thread.java:748)
Caused by: de.fau.cs.osr.utils.visitor.VisitorException: vClass: de.tudarmstadt.ukp.wikipedia.api.sweble.PlainTextConverter
nClass: org.sweble.wikitext.parser.nodes.WtText
Candidate 1: visit(org.sweble.wikitext.parser.nodes.WtNode)
Candidate 2: visit(de.fau.cs.osr.ptk.common.ast.AstText)

	at de.fau.cs.osr.utils.visitor.VisitorLogic.resolveAndVisit(VisitorLogic.java:130)
	at de.fau.cs.osr.utils.visitor.VisitorLogic.resolveAndVisit(VisitorLogic.java:90)
	at de.fau.cs.osr.utils.visitor.VisitorBase.resolveAndVisit(VisitorBase.java:119)
	at de.fau.cs.osr.ptk.common.AstVisitor.dispatch(AstVisitor.java:56)
	at de.fau.cs.osr.ptk.common.AstVisitor.iterate(AstVisitor.java:66)
	at de.tudarmstadt.ukp.wikipedia.api.sweble.PlainTextConverter.visit(PlainTextConverter.java:346)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at de.fau.cs.osr.utils.visitor.VisitorLogic$Target.invoke(VisitorLogic.java:361)
	at de.fau.cs.osr.utils.visitor.VisitorLogic.resolveAndVisit(VisitorLogic.java:110)
	... 79 more
Caused by: vClass: de.tudarmstadt.ukp.wikipedia.api.sweble.PlainTextConverter
nClass: org.sweble.wikitext.parser.nodes.WtText
Candidate 1: visit(org.sweble.wikitext.parser.nodes.WtNode)
Candidate 2: visit(de.fau.cs.osr.ptk.common.ast.AstText)

	at de.fau.cs.osr.utils.visitor.VisitorLogic$1.compare(VisitorLogic.java:186)
	at de.fau.cs.osr.utils.visitor.VisitorLogic$1.compare(VisitorLogic.java:168)
	at java.util.TimSort.countRunAndMakeAscending(TimSort.java:355)
	at java.util.TimSort.sort(TimSort.java:220)
	at java.util.Arrays.sort(Arrays.java:1512)
	at java.util.ArrayList.sort(ArrayList.java:1462)
	at java.util.Collections.sort(Collections.java:175)
	at de.fau.cs.osr.utils.visitor.VisitorLogic.findVisit(VisitorLogic.java:167)
	at de.fau.cs.osr.utils.visitor.VisitorLogic.resolveAndVisit(VisitorLogic.java:105)
	... 90 more
```

It seems there is a mismatch of method signatures and/or incompatible libraries being used at runtime. I consider this a major bug, as parts of the main functionality are affected. Therefore, this bug should be fixed before releasing JWPL 1.2.0 (Final).

Dependencies involved:
-
- swebble 3.1.7
- de.fau.cs.osr.ptk.ptk-common 3.0.8
- de.fau.cs.osr.utils 3.0.8

System environment:
-
- Affected OS: MacOS 10.13, Linux / Ubuntu 16.04 LTS+, most likely Windows 7/8/10, cc: @rzo1
- Java version: Oracle JDK 1.8 (build 1.8.0_161-b12), most likely every 1.8.0-x release
 
Any ideas @ferschke  / @reckart ? Can somebody contact the colleagues at FAU Erlangen to investigate this issue?

**Comment by mawiesne** [comment]
This seems to be a regression introduced with the changes of #152.

**Comment by mawiesne** [comment]
@tgalery as you contributed the changes of #155, can you also have a look into this issue?

**Comment by rzo1** [comment]
I can confirm, that this also affects **Windows 10** - Stacktrace is similar to the one posted by @mawiesne

**Comment by reckart** [comment]
@mawiesne No idea. I hope @tgalery maybe has some insight.

**Comment by rzo1** [comment]
```
    public static void main(String[] args) throws WikiServiceException, WikiApiException {

        Wikipedia wikipedia = new Wikipedia(new JWPLDataSource("host", "dbname", "user", "password", "com.mysql.jdbc.Driver", false));

        //German Wikipedia for example, page with title "Gesundheit"
        Page page = wikipedia.getPage("Gesundheit");

        //Exception will be thrown...
        page.getPlainText();

    }
```

**Comment by tgalery** [comment]
Can you help me understand something in your code ? Looking at the `JWPLDataSource` would that connect to a database which contains the relevant wikipages ? The creds look funny to me.

**Comment by rzo1** [comment]
Basically:

1.) Create a connection to a database. In our case: a MySQL DB containing the Wikipedia Dumps and therefore the wikipedia pages. 

2.) I left out the real credentials ;)

3.) Retrieve a page of interest (it does not matter which one).

4.) Try to retrieve the full text via `getPlainText()`

**Comment by tgalery** [comment]
gotcha, sorry for being a pain, cause i use this in the context of json wikipedia. Is the Mysql database populated by downloading and importing sql files from here https://dumps.wikimedia.org/enwiki/20180320/ (if so could you let me know which) or is there a transformation from the full xml dump into sql that is done by some cli tool in advance

**Comment by mawiesne** [comment]
We make use of the `DataMachine` tool, provided by JWPL project, see here: https://dkpro.github.io/dkpro-jwpl/DataMachine/

The resulting files are then imported into a MySQL 5.7 installation.

**Comment by tgalery** [comment]
Cool, could I get the exact command you guys used to produce the german (or any other language) dump (I will try to replicate the bug and see if there's an easy fix).

**Comment by rzo1** [comment]
I updated the code-snippet above to not use internal classes / provided related code to execute it.

**Comment by mawiesne** [comment]
@tgalery Thanks a ton for looking into this!

**Comment by tgalery** [comment]
@mawiesne that would be extremely helpful

**Comment by mawiesne** [comment]
@tgalery Download one or both of the two mysql dumps from here:

1. German version (4.5G): 
https://download.mi.hs-heilbronn.de/tulum/wikipedia_de_jwpl_Jan2018.sql.gz
2. Spanish version (2.7G): 
https://download.mi.hs-heilbronn.de/tulum/wikipedia_es_jwpl_Jan2018.sql.gz

Re-Import them on your local dev-machine via:

1. In a MySQL shell/tool: 
`CREATE DATABASE wikipedia_de_jwpl_Jan2018 CHARACTER SET UTF8;`
2. In a MySQL shell/tool: 
`GRANT ALL ON wikipedia_de_jwpl_Jan2018.* TO username@'%' IDENTIFIED BY "password";`
3. From a command line/shell: 
`gunzip < wikipedia_de_jwpl_Jan2018.sql.gz | mysql --quick --user=root -p`

Same procedure with smaller Spanish (_es_) version, just exchange 'de' with 'es'. When you decide to use _es_, you could, for instance, fetch a page such as "Salud".

**Comment by tgalery** [comment]
Cheers, will give you guys an update as soon as I can.

**Comment by tgalery** [comment]
Some upates. I've been trying to debug this using the Spanish dump as it's slightly smaller.
But it seems I get an exception instantiating the wikipedia class. I'm using scala and I get the following:

```
scala> import de.tudarmstadt.ukp.wikipedia.api._
import de.tudarmstadt.ukp.wikipedia.api._

scala> val source = new CustomDataSource("host", "dbname", "user", "password", "com.mysql.jdbc.Driver", false)
SLF4J: Failed to load class "org.slf4j.impl.StaticLoggerBinder".
SLF4J: Defaulting to no-operation (NOP) logger implementation
SLF4J: See http://www.slf4j.org/codes.html#StaticLoggerBinder for further details.
source: de.tudarmstadt.ukp.wikipedia.api.CustomDataSource = de.tudarmstadt.ukp.wikipedia.api.CustomDataSource@3ac02398

scala> val wikipedia = new Wikipedia(source)
log4j:WARN No appenders could be found for logger (de.tudarmstadt.ukp.wikipedia.api.Wikipedia).
log4j:WARN Please initialize the log4j system properly.
log4j:WARN See http://logging.apache.org/log4j/1.2/faq.html#noconfig for more info.
org.hibernate.tool.schema.spi.SchemaManagementException: Schema-validation: missing column [version] in table [MetaData]
  at org.hibernate.tool.schema.internal.AbstractSchemaValidator.validateTable(AbstractSchemaValidator.java:136)
  at org.hibernate.tool.schema.internal.GroupedSchemaValidatorImpl.validateTables(GroupedSchemaValidatorImpl.java:42)
  at org.hibernate.tool.schema.internal.AbstractSchemaValidator.performValidation(AbstractSchemaValidator.java:89)
  at org.hibernate.tool.schema.internal.AbstractSchemaValidator.doValidation(AbstractSchemaValidator.java:68)
  at org.hibernate.tool.schema.spi.SchemaManagementToolCoordinator.performDatabaseAction(SchemaManagementToolCoordinator.java:191)
  at org.hibernate.tool.schema.spi.SchemaManagementToolCoordinator.process(SchemaManagementToolCoordinator.java:72)
  at org.hibernate.internal.SessionFactoryImpl.<init>(SessionFactoryImpl.java:312)
  at org.hibernate.boot.internal.SessionFactoryBuilderImpl.build(SessionFactoryBuilderImpl.java:462)
  at org.hibernate.cfg.Configuration.buildSessionFactory(Configuration.java:710)
  at de.tudarmstadt.ukp.wikipedia.api.hibernate.WikiHibernateUtil.getSessionFactory(WikiHibernateUtil.java:51)
  at de.tudarmstadt.ukp.wikipedia.api.Wikipedia.__getHibernateSession(Wikipedia.java:761)
  at de.tudarmstadt.ukp.wikipedia.api.MetaData.<init>(MetaData.java:44)
  at de.tudarmstadt.ukp.wikipedia.api.Wikipedia.<init>(Wikipedia.java:87)
  ... 42 elided
```

Is there something wrong with the spanish dump I downloaded above ?

**Comment by tgalery** [comment]
Commenting out hibernate auto validation gives me this:

```
Caused by: com.mysql.jdbc.exceptions.jdbc4.MySQLSyntaxErrorException: Unknown column 'metadata0_.version' in 'field list'
  at sun.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
  at sun.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62)
  at sun.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
  at java.lang.reflect.Constructor.newInstance(Constructor.java:423)
  at com.mysql.jdbc.Util.handleNewInstance(Util.java:411)
  at com.mysql.jdbc.Util.getInstance(Util.java:386)
  at com.mysql.jdbc.SQLError.createSQLException(SQLError.java:1052)
  at com.mysql.jdbc.MysqlIO.checkErrorPacket(MysqlIO.java:3597)
  at com.mysql.jdbc.MysqlIO.checkErrorPacket(MysqlIO.java:3529)
  at com.mysql.jdbc.MysqlIO.sendCommand(MysqlIO.java:1990)
  at com.mysql.jdbc.MysqlIO.sqlQueryDirect(MysqlIO.java:2151)
  at com.mysql.jdbc.ConnectionImpl.execSQL(ConnectionImpl.java:2625)
  at com.mysql.jdbc.PreparedStatement.executeInternal(PreparedStatement.java:2119)
  at com.mysql.jdbc.PreparedStatement.executeQuery(PreparedStatement.java:2281)
  at org.hibernate.engine.jdbc.internal.ResultSetReturnImpl.extract(ResultSetReturnImpl.java:60)
  ... 60 more

```

so ... maybe there's something funny with the dump ?

**Comment by mawiesne** [comment]
@tgalery I think I know what went wrong, and I'll provide two modified/fresh dumps on next Monday. Sry, for any inconveniences.

**Comment by mawiesne** [comment]
@tgalery Re-Download both files and check _sha1sum_ afterwards:

1. German version (4.5G): 
https://download.mi.hs-heilbronn.de/tulum/wikipedia_de_jwpl_Jan2018.sql.gz
_sha1sum_should match `f837788b0fe5c5b564fd22f11213be9d718190f4`

2. Spanish version (2.7G): 
https://download.mi.hs-heilbronn.de/tulum/wikipedia_es_jwpl_Jan2018.sql.gz
_sha1sum_ should match `dc33b2975e4243217e13658685de2bcf3677975a`

Remove all previous files / imported DBs and conduct a re-import. It should work now as I've dumped it from one of our production systems in which `no` DB schema errors are present.

**Comment by rzo1** [comment]
It seems to be a problem with the reflection code in `de.fau.cs.osr.utils.visitor.VisitorLogic`, which cannot differentiate between the correct `visit` methods at runtime...

**Comment by rzo1** [comment]
At Heilbronn University Group we managed to reproduce this bug with the existing test-cases and the test-DB provided in #2, see

```
org.junit.internal.AssumptionViolatedException: got: <de.fau.cs.osr.utils.visitor.VisitingException: de.fau.cs.osr.utils.visitor.VisitorException: vClass: de.tudarmstadt.ukp.wikipedia.api.sweble.PlainTextConverter
nClass: org.sweble.wikitext.parser.nodes.WtText
Candidate 1: visit(org.sweble.wikitext.parser.nodes.WtNode)
Candidate 2: visit(de.fau.cs.osr.ptk.common.ast.AstText)
>, expected: null


	at org.junit.Assume.assumeThat(Assume.java:95)
	at org.junit.Assume.assumeNoException(Assume.java:142)
	at de.tudarmstadt.ukp.wikipedia.api.PageTest.testPlainText(PageTest.java:100)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:47)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:44)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:271)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:70)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:50)
	at org.junit.runners.ParentRunner$3.run(ParentRunner.java:238)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:63)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:236)
	at org.junit.runners.ParentRunner.access$000(ParentRunner.java:53)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:229)
	at org.junit.internal.runners.statements.RunBefores.evaluate(RunBefores.java:26)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:309)
	at org.junit.runner.JUnitCore.run(JUnitCore.java:160)
	at com.intellij.junit4.JUnit4IdeaTestRunner.startRunnerWithArgs(JUnit4IdeaTestRunner.java:68)
	at com.intellij.rt.execution.junit.IdeaTestRunner$Repeater.startRunnerWithArgs(IdeaTestRunner.java:47)
	at com.intellij.rt.execution.junit.JUnitStarter.prepareStreamsAndStart(JUnitStarter.java:242)
	at com.intellij.rt.execution.junit.JUnitStarter.main(JUnitStarter.java:70)
Caused by: de.fau.cs.osr.utils.visitor.VisitingException: de.fau.cs.osr.utils.visitor.VisitorException: vClass: de.tudarmstadt.ukp.wikipedia.api.sweble.PlainTextConverter
nClass: org.sweble.wikitext.parser.nodes.WtText
Candidate 1: visit(org.sweble.wikitext.parser.nodes.WtNode)
Candidate 2: visit(de.fau.cs.osr.ptk.common.ast.AstText)

	at de.fau.cs.osr.utils.visitor.VisitorBase.handleVisitingException(VisitorBase.java:92)
	at de.fau.cs.osr.utils.visitor.VisitorLogic.resolveAndVisit(VisitorLogic.java:118)
	at de.fau.cs.osr.utils.visitor.VisitorLogic.resolveAndVisit(VisitorLogic.java:90)
	at de.fau.cs.osr.utils.visitor.VisitorBase.resolveAndVisit(VisitorBase.java:119)
	at de.fau.cs.osr.ptk.common.AstVisitor.dispatch(AstVisitor.java:56)
	at de.fau.cs.osr.ptk.common.AstVisitor.iterate(AstVisitor.java:66)
	at de.tudarmstadt.ukp.wikipedia.api.sweble.PlainTextConverter.visit(PlainTextConverter.java:346)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at de.fau.cs.osr.utils.visitor.VisitorLogic$Target.invoke(VisitorLogic.java:361)
	at de.fau.cs.osr.utils.visitor.VisitorLogic.resolveAndVisit(VisitorLogic.java:110)
	at de.fau.cs.osr.utils.visitor.VisitorLogic.resolveAndVisit(VisitorLogic.java:90)
	at de.fau.cs.osr.utils.visitor.VisitorBase.resolveAndVisit(VisitorBase.java:119)
	at de.fau.cs.osr.ptk.common.AstVisitor.dispatch(AstVisitor.java:56)
	at de.fau.cs.osr.ptk.common.AstVisitor.iterate(AstVisitor.java:66)
	at de.tudarmstadt.ukp.wikipedia.api.sweble.PlainTextConverter.visit(PlainTextConverter.java:189)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at de.fau.cs.osr.utils.visitor.VisitorLogic$Target.invoke(VisitorLogic.java:361)
	at de.fau.cs.osr.utils.visitor.VisitorLogic.resolveAndVisit(VisitorLogic.java:110)
	at de.fau.cs.osr.utils.visitor.VisitorLogic.resolveAndVisit(VisitorLogic.java:90)
	at de.fau.cs.osr.utils.visitor.VisitorBase.resolveAndVisit(VisitorBase.java:119)
	at de.fau.cs.osr.ptk.common.AstVisitor.dispatch(AstVisitor.java:56)
	at de.fau.cs.osr.ptk.common.AstVisitor.dispatch(AstVisitor.java:28)
	at de.fau.cs.osr.utils.visitor.VisitorBase.go(VisitorBase.java:111)
	at de.tudarmstadt.ukp.wikipedia.api.Page.parsePage(Page.java:610)
	at de.tudarmstadt.ukp.wikipedia.api.Page.getPlainText(Page.java:591)
	at de.tudarmstadt.ukp.wikipedia.api.PageTest.testPlainText(PageTest.java:98)
	... 23 more
Caused by: de.fau.cs.osr.utils.visitor.VisitorException: vClass: de.tudarmstadt.ukp.wikipedia.api.sweble.PlainTextConverter
nClass: org.sweble.wikitext.parser.nodes.WtText
Candidate 1: visit(org.sweble.wikitext.parser.nodes.WtNode)
Candidate 2: visit(de.fau.cs.osr.ptk.common.ast.AstText)

	at de.fau.cs.osr.utils.visitor.VisitorLogic.resolveAndVisit(VisitorLogic.java:130)
	at de.fau.cs.osr.utils.visitor.VisitorLogic.resolveAndVisit(VisitorLogic.java:90)
	at de.fau.cs.osr.utils.visitor.VisitorBase.resolveAndVisit(VisitorBase.java:119)
	at de.fau.cs.osr.ptk.common.AstVisitor.dispatch(AstVisitor.java:56)
	at de.fau.cs.osr.ptk.common.AstVisitor.iterate(AstVisitor.java:66)
	at de.tudarmstadt.ukp.wikipedia.api.sweble.PlainTextConverter.visit(PlainTextConverter.java:210)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at de.fau.cs.osr.utils.visitor.VisitorLogic$Target.invoke(VisitorLogic.java:361)
	at de.fau.cs.osr.utils.visitor.VisitorLogic.resolveAndVisit(VisitorLogic.java:110)
	... 53 more
Caused by: vClass: de.tudarmstadt.ukp.wikipedia.api.sweble.PlainTextConverter
nClass: org.sweble.wikitext.parser.nodes.WtText
Candidate 1: visit(org.sweble.wikitext.parser.nodes.WtNode)
Candidate 2: visit(de.fau.cs.osr.ptk.common.ast.AstText)

	at de.fau.cs.osr.utils.visitor.VisitorLogic$1.compare(VisitorLogic.java:186)
	at de.fau.cs.osr.utils.visitor.VisitorLogic$1.compare(VisitorLogic.java:168)
	at java.util.TimSort.countRunAndMakeAscending(TimSort.java:355)
	at java.util.TimSort.sort(TimSort.java:220)
	at java.util.Arrays.sort(Arrays.java:1512)
	at java.util.ArrayList.sort(ArrayList.java:1462)
	at java.util.Collections.sort(Collections.java:175)
	at de.fau.cs.osr.utils.visitor.VisitorLogic.findVisit(VisitorLogic.java:167)
	at de.fau.cs.osr.utils.visitor.VisitorLogic.resolveAndVisit(VisitorLogic.java:105)
	... 64 more
```

CI did not complain because auf #161

**Comment by tgalery** [comment]
Cool, I'm assuming this will be reproducible once #162 gets merged ?

**Comment by rzo1** [comment]
Yes

**Comment by rzo1** [comment]
@tgalery Any updates here? :)

**Comment by mawiesne** [comment]
@rzo1 @tgalery Seems, I found a fix for this issue locally. I will push a branch and open a PR, once the related test case works as expected.

**Comment by mawiesne** [comment]
Finally fixed via PR #185
